### PR TITLE
#460: Ignoring backticks

### PR DESCRIPTION
**Pull request by kober32**
#460 Added util function for removing backticks (`) from variables and constants.

Using this function on `[constant-naming]`, `[lower-camel-case]`, and `[constant-k-prefix]` rules

### CODE REVIEW

File: src/test/java/com/sleekbyte/tailor/utils/CharFormatUtilTest.java [hunk 1]
```diff
@@ -88,4 +88,13 @@ public void testStartsWithAcronym() {
         assertFalse(CharFormatUtil.startsWithAcronym("xURLS"));
         assertFalse(CharFormatUtil.startsWithAcronym("shieldPROGRAMMEmarvel"));
     }
+
+    @Test
+    public void testBacktickEscapedIdentifier() {
+        // Backticks are not part of the identifier
```
Comment (adityatrivedi) [inline src/test/java/com/sleekbyte/tailor/utils/CharFormatUtilTest.java]: Maybe we could also test inputs like ``, single character strings, and empty strings to exercise all execution paths?

File: src/main/java/com/sleekbyte/tailor/listeners/LowerCamelCaseListener.java [hunk 2]
```diff
@@ -28,28 +28,29 @@ public LowerCamelCaseListener(Printer printer) {
     @Override
     public void enterTopLevel(TopLevelContext topLevelCtx) {
         List<IdentifierContext> names = DeclarationListener.getVariableNames(topLevelCtx);
-        names.forEach(ctx -> verifyLowerCamelCase(Messages.VARIABLE + Messages.NAMES, ctx));
+        names.forEach(ctx -> verifyLowerCamelCase(Messages.VARIABLE + Messages.NAMES, ctx, true));
```
Comment (adityatrivedi) [inline src/main/java/com/sleekbyte/tailor/listeners/LowerCamelCaseListener.java]: Can you please add some representative functional tests to `ConstantNamingTest.swift` and  `LowerCamelCaseTest.swift`?

File: src/main/java/com/sleekbyte/tailor/listeners/LowerCamelCaseListener.java [hunk 3]
```diff
@@ -28,28 +28,29 @@ public LowerCamelCaseListener(Printer printer) {
...
 
     @Override
     public void enterUnionStyleEnumCase(SwiftParser.UnionStyleEnumCaseContext ctx) {
-        verifyLowerCamelCase(Messages.ENUM_CASE + Messages.NAMES, ctx.enumCaseName());
+        verifyLowerCamelCase(Messages.ENUM_CASE + Messages.NAMES, ctx.enumCaseName(), false);
     }
 
-    private void verifyLowerCamelCase(String constructType, ParserRuleContext ctx) {
-        String constructName = ctx.getText();
+    private void verifyLowerCamelCase(String constructType, ParserRuleContext ctx, Boolean unescapeIdentifier) {
```
Comment (a4sriniv) [inline src/main/java/com/sleekbyte/tailor/listeners/LowerCamelCaseListener.java]: considering `unescapeIdentifier` is only `true` for variable and constant names, could you please make an overloaded function `verifyLowerCamelCase(constructType, ctx)` that calls this function with `unescapeIdentifier` set to `false` (so we don't have to keep passing in `false`).

**Comment by kober32** [comment]
Hi guys. Thanks for your comments. I will update the PR asap.

j.

**Comment by adityatrivedi** [comment]
Closing this PR and tracking changes under #477.
